nasbench/decoder.py

```diff
@@ -73,25 +75,40 @@ def __init__(self,
         for i in range(self.n):
             self.offsets.append( (i + 3) * i // 2 - 1)
     
-    def forward(self, x, encoder_hidden=None, encoder_outputs=None):
+    def forward(self, x, x_len, encoder_hidden=None, encoder_outputs=None):
```

Comment:
x_len feels like a somewhat misnomer. If i understood it correctly, it is somewhat like this: 
x_len = [len(x) for x in xs]
So, maybe x_len_per_elem? x_len_list? Or at least some comments would be helpful!